src/test/java/com/conveyal/datatools/manager/controllers/api/UserControllerTest.java

```diff
@@ -113,13 +116,196 @@ public void canListFirstTenUsers() throws IOException {
         assertThat(userResponse, matchesSnapshot());
     }
 
+    /**
+     * When creating a new datatools user which matches an existing Auth0 user, make sure the permissions for the
+     * existing Auth0 users can be updated.
+     */
+    @Test
+    public void canUpdatePermissionsOfExistingAuth0User() throws IOException {
+        String newUserEmail = "test-existing-user@test.com";
+
+        // create wiremock stub for user search that matches existing user
+        String url = USERS_API_PATH + "?sort=email%3A1&per_page=10&page=0&include_totals=false&search_engine=v3&q=email%3A" + URLEncoder.encode(newUserEmail, "UTF-8") + "*";
+        stubFor(
+            get(urlEqualTo(url))
+                .willReturn(
+                    aResponse()
+                        .withBodyFile("userSearchResponse.json")
+                )
+        );
+
+        // create wiremock stub for update users endpoint
+        stubFor(
+            patch(urlPathEqualTo(USERS_API_PATH + "/auth0%7Ctest-existing-user"))
+                .withRequestBody(
+                    matchingJsonPath(
+                        "$.app_metadata.datatools[0].permissions[0].type",
+                        equalTo("administer-application")
+                    )
+                )
+                .willReturn(
+                    aResponse()
+                        .withBodyFile("updateExistingUserResponse.json")
+                )
+        );
+
+        ObjectNode requestJson = mapper.createObjectNode();
+        requestJson.put("email", emailForExistingAccount);
+        ObjectNode testClientPermissions = mapper.createObjectNode();
+        testClientPermissions.put("type", "administer-application");
+        ObjectNode testClientData = mapper.createObjectNode();
+        testClientData.putArray("permissions").add(testClientPermissions);
+        testClientData.putArray("projects");
+        testClientData.putArray("organizations");
+        testClientData.put("client_id", "testing-client-id");
+        requestJson.set("permissions", testClientData);
```

Comment:
A lot of this construction of JSON looks to be repeated with other tests. Opportunity to factor this out?

Reply:
Updated.